src/main.ts

```diff
@@ -1,16 +1,135 @@
 import * as core from '@actions/core'
-import {wait} from './wait'
+import * as github from '@actions/github'
+import {Action, TerraformPlan} from './types'
+
+const commentPrefix = '## Terraform Plan:'
 
 async function run(): Promise<void> {
   try {
-    const ms: string = core.getInput('milliseconds')
-    core.debug(`Waiting ${ms} milliseconds ...`)
+    core.debug('got inside the action')
+
+    const pr = github.context.payload.pull_request
+    if (!pr) {
+      core.info('No pull request, exiting action...')
+      return
+    }
+    core.debug('got pull request')
+
+    const terraformPlan: TerraformPlan = JSON.parse(
+      core.getInput('terraform_plan_json')
+    )
+
+    const token = core.getInput('github_token')
+    const nwo = process.env['GITHUB_REPOSITORY'] || '/'
+    const [owner, repo] = nwo.split('/')
+    const octokit = github.getOctokit(token)
+
+    const toCreate = []
+    const toDelete = []
+    const toReplace = []
+    const toUpdate = []
+    for (const resourceChange of terraformPlan.resource_changes) {
+      core.debug(`resource: ${JSON.stringify(resourceChange)}`)
+      const actions = resourceChange.change.actions
+      const resourceName = `${resourceChange.type} - ${resourceChange.name}`
+      if (actions.length === 1 && actions.includes(Action.create)) {
+        toCreate.push(resourceName)
+      } else if (actions.length === 1 && actions.includes(Action.delete)) {
+        toDelete.push(resourceName)
+      } else if (
+        actions.length === 2 &&
+        actions.includes(Action.delete) &&
+        actions.includes(Action.create)
+      ) {
+        toReplace.push(resourceName)
+      } else if (actions.length === 1 && actions.includes(Action.update)) {
+        toUpdate.push(resourceName)
+      } else {
+        core.debug(`Not found? ${actions}`)
+      }
+    }
+
+    core.debug(`toCreate: ${toCreate}`)
+    core.debug(`toUpdate: ${toUpdate}`)
+    core.debug(`toReplace: ${toReplace}`)
+    core.debug(`toDelete: ${toDelete}`)
 
-    core.debug(new Date().toTimeString())
-    await wait(parseInt(ms, 10))
-    core.debug(new Date().toTimeString())
+    let body = `${commentPrefix}\n`
+    if (toCreate.length > 0) {
+      body += `will create ${toCreate.length} resources: \n`
```

Comment:
Nit:
```suggestion
      body += `will create ${toCreate.length} resource${toCreate.length > 1 ? 's' : ''}: \n`
```

Reply:
This applies to each of the 4 nearly-identical sections. It fixes 
>will **delete** 1 resources:

Reply:
Refactored into a class with more reusable pieces of code